pkg/authz/tenant/service.go

```diff
@@ -144,6 +150,10 @@ func (svc TenantService) DeleteByTenantId(ctx context.Context, tenantId string)
 			return err
 		}
 
+		err = event.NewService(svc.Env()).TrackResourceDeletedSync(ctx, ResourceTypeTenant, tenantId, nil)
```

Comment:
Why is this using the `TrackResourceDeletedSync` method instead of the async delete used in the other resources?